java/src/test/scala/com/example/JavaOtelSuite.scala

```diff
@@ -0,0 +1,193 @@
+package com.example
+
+import cats.effect.{IO, Sync}
+import io.opentelemetry.sdk.OpenTelemetrySdk
+import io.opentelemetry.sdk.metrics._
+import io.opentelemetry.sdk.metrics.data.{
+  HistogramPointData,
+  MetricDataType,
+  SummaryPointData,
+  MetricData => JMetricData
+}
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricReader
+import munit.CatsEffectSuite
+import org.typelevel.otel4s.java.OtelJava
+import org.typelevel.otel4s.{Attribute, AttributeKey, Otel4s}
+
+import scala.jdk.CollectionConverters._
+
+class JavaOtelSuite extends CatsEffectSuite {
+  import JavaOtelSuite._
+
+  test("Counter record a proper data") {
+    val expected = Metric(
+      "counter",
+      Some("description"),
+      Some("unit"),
+      MetricData.LongSum(List(1))
+    )
+
+    for {
+      sdk <- IO.delay(makeSdk)
+      meter <- sdk.otel4s.meterProvider.get("java.otel.suite")
+
+      counter <- meter
+        .counter("counter")
+        .withUnit("unit")
+        .withDescription("description")
+        .create
+
+      _ <- counter.add(
+        1L,
+        Attribute(AttributeKey.string("string-attribute"), "value")
+      )
+
+      metrics <- sdk.metrics
+    } yield assertEquals(metrics, List(expected))
+  }
+
+  private def makeSdk: OtelSdk[IO] = {
+    val metricReader = InMemoryMetricReader.create()
+
+    val meterProvider = SdkMeterProvider
+      .builder()
+      .registerMetricReader(metricReader)
+      .registerView(
+        InstrumentSelector.builder().setType(InstrumentType.HISTOGRAM).build(),
+        View
+          .builder()
+          .setAggregation(
+            Aggregation
+              .explicitBucketHistogram(HistogramBuckets.map(Double.box).asJava)
+          )
+          .build()
+      )
+      .build()
+
+    val sdk = OpenTelemetrySdk
+      .builder()
+      .setMeterProvider(meterProvider)
+      .build()
+
+    new OtelSdk[IO](sdk, metricReader)
+  }
+
+  private val HistogramBuckets: List[Double] =
+    List(0.01, 1, 100)
+
+}
+
+object JavaOtelSuite {
```

Comment:
Would this object be useful in a testkit module?

Reply:
Yes. I moved it to the `testkit` module. You can review it.